### ISSUE TITLE
remove BOM entry not working

### ISSUE THREAD

**Issue by oceanofthelost** [oceanofthelost/InteractivePCB#69]
Adding any value to "Remove BOM Entry" will not remove the value from the BOM table. 

Error is in function
```js
function filterBOM_ByAttribute(part)
{
    let result = false;
    let splitFilterString = globalData.getRemoveBOMEntries().split(",");
    // Remove null, "", undefined, and 0 values
    splitFilterString    = splitFilterString.filter(function(e){return e});

    if(splitFilterString.length > 0 )
    {
        for(let i of splitFilterString)
        {
            // removing beginning and trailing whitespace
            i = i.trim()
            if(part.attributes.has(i))
            {
                // Id the value is an empty string then dont filter out the entry. 
                // if the value is anything then filter out the bom entry
                if(part.attributes.get(i) != "")
                {
                    result = true;
                }
            }
        }
    }
    return result;
}
```

specifically the line

```js
 if(part.attributes.has(i))
```
using ```has()``` on a Map will perform a key search where what should be done is a search of each keys value.

**Comment by oceanofthelost** [comment]
Issue resolved in fe7b150c378e8fc68e62523c5763b038625774e2. 

Note that filtering function is **STRICT**, meaning the filter string must exactly match the attribute string, otherwise the BOM entry will not be filtered out. 

This behavior is by design as a user may want to have attributes that are similar but differ in case and this makes sure that only what the user wants filtered is removed.

**Comment by oceanofthelost** [comment]
Added to release [V2.3.0](https://github.com/oceanofthelost/InteractivePCB/releases/tag/V2.3.0)
